uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -27,9 +27,11 @@ define([
     ) {
       $log.debug('RequestCtrl');
       var absenceTypesAndIds;
+      var availableStatusesMatrix = {};
       var initialLeaveRequestAttributes = {}; // used to compare the change in leaverequest in edit mode
       var role = '';
       var NO_ENTITLEMENT_ERROR = 'No entitlement';
+      var vm;
```

Comment:
In *this particular file* we have to use "this" and "self" instead of "vm" at the moment (I am not sure we want to mix vm and this, it is one or another, or you will need to refactor the whole file). Please define `this. requestStatuses = {};` at the start and all similar props, that would enhance the readability.